### ISSUE TITLE
`React.useEffect` for HTTP requests

### ISSUE THREAD

**Issue by MG10327** [MG10327/Documentation#149]
HTTP Requests are common in React.

HTTP requests are made with `React.useEffect()`.

It's a `hook` `callback` so that means it takes another function as an argument. It also means you will use `async` / `await` syntax.

Although you can not return anything from `React.useEffect()` other than the clean up function.

The cleanup function is a function in the `useEffect` hook that allows us to tidy up our code before our component unmounts. 

Every time the code renders and re-renders `useEffect` also cleans itself up with the cleanup function.

For Example:
```
// this does not work, don't do this:
React.useEffect(async () => {
  const result = await doSomeAsyncThing()
  // do something with the result
})
```

On the other hand, this will ONLY return the clean up function because `async` will make the function inside the React Hook wait until the `promise` is fulfilled to start working

```
React.useEffect(() => {
  async function effect() {
    const result = await doSomeAsyncThing()
    // do something with the result
  }
  effect()
})
```
In short, put your `async` function INSIDE the `React.useEffect()` hook. Do not turn the `React.useEffect()` Hook INTO an `async` function like the first example. Notice the difference in the `async` placement.

The basic makeup of a `React.useEffect()` call is the following:

```
React.useEffect(() => {
// code to execute, [dependencies]

```

Every time the state of the dependency changes your component will re-render. If you don't write a dependency array at all then your component will re-render every time the state of anything on the page changes.
